tests/pytests/test_config.py

```diff
@@ -370,14 +371,45 @@ def testAllConfigCoord(env):
 
 @skip(cluster=False)
 def testInitConfigCoord():
-    def test_arg_num(arg_name, arg_value):
-        env = Env(moduleArgs=f'{arg_name} {arg_value}', noDefaultModuleArgs=True)
-        env.expect(config_cmd(), 'get', arg_name).equal([[arg_name, str(arg_value)]])
+
+    _test_arg_num('SEARCH_THREADS', 3)
+    _test_arg_num('CONN_PER_SHARD', 3)
+
+    def _testOSSGlobalPasswordConfig():
+        env = Env(moduleArgs='OSS_GLOBAL_PASSWORD 123456', noDefaultModuleArgs=True)
+        if env.env == 'existing-env':
+            env.skip()
+        env.expect(config_cmd(), 'get', 'OSS_GLOBAL_PASSWORD').equal([['OSS_GLOBAL_PASSWORD', 'Password: *******']])
         env.stop()
 
-    test_arg_num('SEARCH_THREADS', 3)
-    test_arg_num('CONN_PER_SHARD', 3)
+    # We test `OSS_GLOBAL_PASSWORD` manually since the getter obfuscates the value
+    _testOSSGlobalPasswordConfig()
+
+    _test_arg_str('OSS_ACL_USERNAME', 'default')
 
 @skip(cluster=False)
 def testImmutableCoord(env):
     env.expect(config_cmd(), 'set', 'SEARCH_THREADS').error().contains(not_modifiable)
+    env.expect(config_cmd(), 'set', 'OSS_GLOBAL_PASSWORD').error().contains(not_modifiable)
+    env.expect(config_cmd(), 'set', 'OSS_ACL_USERNAME').error().contains(not_modifiable)
+
+@skip(cluster=False)
+def testSetACLUsername():
+    """Tests that the OSS_ACL_USERNAME configuration is set correctly on module
+    load"""
+
+    # Setting the `OSS_ACL_USERNAME` configuration without the `OSS_GLOBAL_PASSWORD`
+    # the configuration should not do anything since we don't try to authenticate.
+    _test_arg_str('OSS_ACL_USERNAME', 'test')
+
+    # Set both the username and password. This should fail since we have no such
+    # user.
+    env = Env(moduleArgs='OSS_ACL_USERNAME test_user OSS_GLOBAL_PASSWORD 123456', noDefaultModuleArgs=True)
+
+    timeout = 3 # 5 seconds, more than enough for the an env to be up
+    try:
+        with TimeLimit(timeout):
```

Comment:
I  think the behavior should be that the user gets a proper error.
I don't think we should be in an infinite loop.

Reply:
I agree in general, but this is an interesting state cluster-wise as well, thus should be thought about separately since this is an existing issue.
Currently the cluster will need a restart for the new load-time config-param values anyways so I'm not sure this is a big problem.
Bottom line - this is the current behavior, and if we want to fix it we need to think about it more and do it in a more complete manner separately.

Reply:
I would accept the existing behavior argument if the test didn't exist.
The fact you created a test to verify this behavior seems to imply this behavior is part of the code you added.
Otherwise what exactly did you add behavior wise that you want to test.